### PR TITLE
Fix CMS deprecation warnings in L1Trigger/L1TCommon

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- converted to thread-friendly module types
- added esConsumes

#### PR validation:

Code compiles without issuing the deprecation warnings.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: L1Trigger/L1TCommon/plugins/L1TConfigDumper.cc [hunk 1]
```diff
@@ -162,168 +162,194 @@
...
 
 L1TConfigDumper::L1TConfigDumper(const edm::ParameterSet& iConfig)
-
-{}
-
-L1TConfigDumper::~L1TConfigDumper() {}
+    : AToken_(esConsumes()),
+      BToken_(esConsumes()),
+      CToken_(esConsumes()),
+      EToken_(esConsumes()),
```
Comment (missirol) [inline L1Trigger/L1TCommon/plugins/L1TConfigDumper.cc:222]:
```suggestion
      CToken_(esConsumes()),
      DToken_(esConsumes()),
      EToken_(esConsumes()),
```
Maybe one call was missed? :)

Comment (Dr15Jones) [reply L1Trigger/L1TCommon/plugins/L1TConfigDumper.cc]: Good catch! Guess I don't know the alphabet :P

Comment (Dr15Jones) [reply L1Trigger/L1TCommon/plugins/L1TConfigDumper.cc]: fixed

**Comment by missirol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by perrotta** [comment]
+1

**Pull request merged**